### PR TITLE
CPLAT-14914 Null-safety and better compatibility with build filtering

### PR DESCRIPTION
**Pull request by evanweible-wf**
Due to the current dependency on build_* package versions prior to the introduction of the `$package$` synthetic asset, there is a problem with consumers who try to use the browser aggregation feature along with `--build-filter` args to target a subset of tests: the `dart_test.browser_aggregate.yaml` file is not included in the outputs because it depends on the `$test$` synthetic asset, which itself doesn't get output if you use build filters. The `$package$` asset behaves differently in that it is always produced (because it represents the entire package), even if you use build filters, which in turn means the `dart_test.browser_aggregate.yaml` file is always written. This means devs can use build filters (or tools that use them) to target subsets of tests without breaking test running by having their `dart_test.yaml` reference an asset that doesn't exist.

A couple of related notes:
- In order to use the `$package$` synthetic asset, we had to upgrade the build_* packages to versions that subsequently required a newer version of the Dart SDK (2.12+).
- It makes a bit more sense to output the `dart_test.browser_aggregate.yaml` file in the root (instead of `test/`) so that it lives alongside the `dart_test.yaml` that must include it. This is a breaking change.

Because of those two things, we're landing these changes along with the migration to null-safety so that we can group them all in a single major version release.

## Code Review

It is probably easier to review this PR commit by commit.

## Testing
- [ ] CI passes
- [ ] Can run `ddev test <path/to/my_test.dart>` with browser aggregation enabled successfully

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: README.md [hunk 1]
```diff
@@ -7,12 +7,7 @@ builder can apply a template to any number of test files.
 
 ## Usage
 
-Add this package as a dev_dependency:
-
-```yaml
-dev_dependencies:
-  test_html_builder: ^1.0.0
-```
+Add this package as a dev_dependency: `dart pub add --dev test_html_builder`
```
Comment (todbachman-wf) [inline README.md:10]: :whoa: neat!

**Review by todbachman-wf — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (evanweible-wf) [reply README.md:10]: Yeah, I can't wait to get on the latest!

**Review by todbachman-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
The build filtering fix has been merged into 2.x so I'm closing this PR and opening a separate one for just the NNBD migration.

**Pull request closed without merging**